client/service/src/config.rs

```diff
@@ -251,3 +253,36 @@ impl std::convert::From<PathBuf> for BasePath {
 		BasePath::new(path)
 	}
 }
+
+/// Callable object that execute tasks.
+#[derive(Clone)]
+pub struct TaskExecutor(Arc<dyn Fn(Pin<Box<dyn Future<Output = ()> + Send>>, TaskType) + Send + Sync>);
+
+impl std::fmt::Debug for TaskExecutor {
+	fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+		write!(f, "TaskExecutor")
+	}
+}
+
+impl std::convert::From<Arc<dyn Fn(Pin<Box<dyn Future<Output = ()> + Send>>, TaskType) + Send + Sync>>
+for TaskExecutor {
+	fn from(x: Arc<dyn Fn(Pin<Box<dyn Future<Output = ()> + Send>>, TaskType) + Send + Sync>)
+	-> Self {
+		Self(x)
+	}
+}
+
+impl std::ops::Deref for TaskExecutor {
+	type Target = Arc<dyn Fn(Pin<Box<dyn Future<Output = ()> + Send>>, TaskType) + Send + Sync>;
+
+	fn deref(&self) -> &Self::Target {
+		&self.0
+	}
+}
+
+impl TaskExecutor {
+	/// Create a `TaskExecutor` from a function
+	pub fn from_fn(f: impl Fn(Pin<Box<dyn Future<Output = ()> + Send>>, TaskType) + Send + Sync + 'static) -> Self {
```

Comment:
Couldn't this also be implemented as `From<>`?

Reply:
Yes!! Thanks it worked great 7b0590e43